src/components/application_manager/src/commands/command_request_impl.cc

```diff
@@ -791,48 +818,9 @@ bool CommandRequestImpl::PrepareResultForMobileResponse(
 bool CommandRequestImpl::PrepareResultForMobileResponse(
     ResponseInfo& out_first, ResponseInfo& out_second) const {
   LOG4CXX_AUTO_TRACE(logger_);
-  using namespace helpers;
-
-  out_first.is_ok = Compare<hmi_apis::Common_Result::eType, EQ, ONE>(
-      out_first.result_code,
-      hmi_apis::Common_Result::SUCCESS,
-      hmi_apis::Common_Result::WARNINGS,
-      hmi_apis::Common_Result::WRONG_LANGUAGE,
-      hmi_apis::Common_Result::RETRY,
-      hmi_apis::Common_Result::SAVED);
-
-  out_second.is_ok = Compare<hmi_apis::Common_Result::eType, EQ, ONE>(
-      out_second.result_code,
-      hmi_apis::Common_Result::SUCCESS,
-      hmi_apis::Common_Result::WARNINGS,
-      hmi_apis::Common_Result::WRONG_LANGUAGE,
-      hmi_apis::Common_Result::RETRY,
-      hmi_apis::Common_Result::SAVED);
-
-  out_first.is_not_used =
-      hmi_apis::Common_Result::INVALID_ENUM == out_first.result_code;
-
-  out_second.is_not_used =
-      hmi_apis::Common_Result::INVALID_ENUM == out_second.result_code;
-
-  out_first.is_unsupported_resource =
-      hmi_apis::Common_Result::UNSUPPORTED_RESOURCE == out_first.result_code;
-
-  out_second.is_unsupported_resource =
-      hmi_apis::Common_Result::UNSUPPORTED_RESOURCE == out_second.result_code;
-
-  out_first.interface_state =
-      application_manager_.hmi_interfaces().GetInterfaceState(
-          out_first.interface);
-
-  out_second.interface_state =
-      application_manager_.hmi_interfaces().GetInterfaceState(
-          out_second.interface);
-
   bool result = (out_first.is_ok && out_second.is_ok) ||
-                (out_second.is_not_used && out_first.is_ok) ||
-                (out_first.is_not_used && out_second.is_ok);
-
+                (out_first.is_ok && out_second.is_not_used) ||
```

Comment:
These checks should likely be in the `CheckResultCode` function instead of here (`CheckResultCode` is confusing as-is)

Something like:

```
bool CheckResultCode(const ResponseInfo& first, const ResponseInfo& second) {
  if (first.is_ok && second.is_ok) {
    return true;
  }
  
  if (first.is_ok && second.is_not_used) {
    return true;
  }

  if (first.is_ok && second.is_unsupported_resource) {
    return true;
  }
  return false;
}
```

```
bool result = CheckResultCode(out_first, out_second) || CheckResultCode(out_second, out_first);
```

Reply:
@jacobkeeler agree, will insert it to CheckResultCode function.

Reply:
@jacobkeeler fixed 931ce35